blog/2025-11-23-mqtt-subscribe-wait.md

```diff
@@ -0,0 +1,90 @@
+---
+author: Jan Bouwhuis
+authorURL: https://github.com/jbouwh
+authorImageURL: https://avatars.githubusercontent.com/u/7188918?s=96&v=4
+title: Add a status callback for MQTT subscriptions
+---
+
+## Add a status callback for MQTT subscriptions
+
+Integrations that use MQTT might need to wait for a subscription to complete before they initiate actions. The default behavior is that a subscription is queued and debounced, so callers usually do not wait for broker confirmation. Some integrations must guarantee that the broker finished the subscription. To support this requirement, the MQTT subscribe API adds a `on_subscribe` callback option.
+
+The new async signature for MQTT subscribe that supports the new callback option is:
+
+```python
+@bind_hass
+async def async_subscribe(
+    hass: HomeAssistant,
+    topic: str,
+    msg_callback: Callable[[ReceiveMessage], Coroutine[Any, Any, None] | None],
+    qos: int = DEFAULT_QOS,
+    encoding: str | None = DEFAULT_ENCODING,
+    on_subscribe: CALLBACK_TYPE,
```

Comment:
We're missing the optional typing and default argument.

https://github.com/home-assistant/core/blob/c8fd6db3ff750306e16bc507ba9cf86b3997a3a6/homeassistant/components/mqtt/client.py#L231